libcloud/compute/drivers/scaleway.py

```diff
@@ -0,0 +1,260 @@
+
+    def parse_error(self):
+        return super(ScalewayResponse, self).parse_error()['message']
+
+    def success(self):
+        return self.status in self.valid_response_codes
+
+
+class ScalewayConnection(ConnectionUserAndKey):
+    host = SCALEWAY_API_HOST
+    allow_insecure = False
+    responseCls = ScalewayResponse
+
+    def add_default_headers(self, headers):
+        headers['X-Auth-Token'] = self.key
+        headers['Content-Type'] = 'application/json'
+        return headers
+
+
+class ScalewayNodeDriver(NodeDriver):
+    type = Provider.SCALEWAY
+    connectionCls = ScalewayConnection
+    name = 'Scaleway'
+    website = 'https://www.scaleway.com/'
+
+    SNAPSHOT_STATE_MAP = {  # TODO map all states
+        'snapshotting': VolumeSnapshotState.CREATING
+    }
+
+    def list_sizes(self):
+        return [NodeSize(driver=self, **copy.deepcopy(size))
+                for size in SCALEWAY_INSTANCE_TYPES]
+
+    def list_images(self):
+        response = self.connection.request('/images')
+        images = response.object['images']
+        return [NodeImage(id=image['id'],
+                          name=image['name'],
+                          driver=self,
+                          extra={'arch': image['arch']})
+                for image in images]
+
+    def create_image(self, node, name, description=None):
+        data = {
+            'organization': self.key,
+            'name': name,
+            'arch': node.extra['arch'],
+            'root_volume': node.extra['volumes']['0']['id']  # TODO check this
+        }
+        response = self.connection.request('/images', data=json.dumps(data),
+                                           method='POST')
+        image = response.object['image']
+        return NodeImage(id=image['id'], name=image['name'], driver=self)
+
+    def delete_image(self, node_image):
+        return self.connection.request('/images/%s' % node_image.id,
+                                       method='DELETE').success()
+
+    def get_image(self, image_id):
+        response = self.connection.request('/images/%s' % image_id)
+        image = response.object['image']
+        return NodeImage(id=image['id'], name=image['name'], driver=self)
+
+    def list_nodes(self):
+        response = self.connection.request('/servers')
+        servers = response.object['servers']
+        return [self._to_node(server) for server in servers]
+
+    def _to_node(self, server):
+        return Node(id=server['id'],
+                    name=server['name'],
+                    state=NodeState.fromstring(server['state']),
+                    public_ips=[server['public_ip']['address']] if server['public_ip'] else [],
+                    private_ips=[server['private_ip']] if server['private_ip'] else [],
+                    driver=self,
+                    extra={'volumes': server['volumes']},
+                    created_at=parse_date(server['creation_date']))
+
+    def create_node(self, **kwargs):
```

Comment:
Please explicitly declare the arguments http://libcloud.readthedocs.org/en/latest/development.html#don-t-abuse-kwargs